OpenRA.Mods.Common/UpdateRules/Rules/20210321/Infiltrates.cs

```diff
@@ -0,0 +1,49 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2022 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+
+namespace OpenRA.Mods.Common.UpdateRules.Rules
+{
+	public class Infiltrates : UpdateRule
+	{
+		public override string Name => "Removes PlayerExperience property from Infiltrates.";
+
+		public override string Description => "Removes PlayerExperience property from Infiltrates and disperses it to all InfiltrateFor Traits.";
+
+		readonly List<string> locations = new List<string>();
+
+		public override IEnumerable<string> AfterUpdate(ModData modData)
+		{
+			if (locations.Count > 0)
+				yield return "The *PlayerExperience fields have been removed from the *Infiltrates traits and\n" +
+				             "added to *InfiltratesFor traits. If you want to keep *PlayerExperience you will\n" +
+							 "need to add it to each of the *InfiltratesFor traits.\n" +
+				             UpdateUtils.FormatMessageList(locations);
```

Comment:
The indentation here seems to be off (and use spaces?)

Reply:
fixed intendation